### DISCUSSION TITLE
WHDLoad game config files - ease of use from GUI?

### DISCUSSION THREAD

**Discussion by celeritydesign** [BlitterStudio/amiberry#921] [Q&A] [answered]
Hi, first of all thank you for Amiberry, it is providing a lot of joy in our house!
I'm posting here as my issues are likely user error, not bugs, so please be kind...

I am running Amiberry 5.0 (2022-03-23) on Retropie 4.8 (I had problems building from source, in the end I had to manually copy capsimg.so to /home/pi/RetroPie-Setup/tmp/build/amiberry/external/capsimg/ - but that is a Retropie issue).

My main query is the best way to save game specific information when running a WHDLoad .lha game directly from Emulation Station.
I can't see an easy way in the Amiberry GUI to save a .uae config file for the current filename.  I know I can type it in manually on the Configurations tab, but there are two issues with this:

1: the filename is often long and difficult to remember.
2: the filename often has special characters like an underscore, and I cannot seem to enter any "shifted" punctuation characters using the official Raspberry Pi keyboard.  Shift + a letter gives me a capital, but shift + a number of any other key gives me the unshifted character. Is this a known issue or something wrong with my setup?

So what I am currently doing, is saving my desired config to a pre-made config, eg "temp.uae", then exiting Emulation Station and manually cloning the config to the correct name to match the game .lha file.  (a use case might be setting up the mouse to work correctly for Amegas).

There is surely a better way than this, I just can't seem to work out what it is :-)

I have tried saving a blank config, in the hope it would default to the game filename, but that didn't work.  Is there a button I'm missing to easily save a game config file with the correct name please?

Cheers!

**Comment by midwan** [comment, marked as answer]
Doesn't it work if you:
- Load the .lha game you want first
- Open the GUI, go to Configurations
- The game-based filename should already be filled in there for you, all you have to do is click on Save. Or make any changes you want first, and then Save, obviously.

**Comment by celeritydesign** [comment]
The configuration textbox is not pre-populated with the game name if I launch the game directly from EmulationStation. It is just set to "default".  This was the same in Amiberry 3.3 (the retropie install from binary option).

**Reply by midwan** [reply]
That sounds weird, do you already have a `default` config there that gets pre-loaded perhaps?
Normally it should show the game title there, so you can click on Save. And it does so for me when I test it here, but I'll need to check RetroPie again separately to be sure.

Regarding the shifted numbers/symbols that didn't work in Textfields, that's a limitation of the GUI toolkit. But now that you mentioned it, I'm preparing an update that will fix that for Amiberry.

**Comment by celeritydesign** [comment]
Yes, I have a default.uae in amiberry/conf but it doesn’t get loaded when I launch a WHDLoad game. I know this because I set it to show scanlines, but no scanlines appear when I load new games.

It does get pre-selected in Configuration though, even though I have multiple configs in the list.  I.e. I load a new game from ES, and amiberry Configuration tab shows:
Name: default 
Description: AutoBoot Configuration [WHDLoad]

If I make test changes to default.uae, eg turn off centering and verify this affects the screen, the changes are definitely not loaded in the next time I load a game.
But, if I then copy default.uae to say, RickDangerous.uae, those changes are honoured when I load the game.

Is there a log I can turn on somewhere that would help diagnose why the config name isn’t being auto-populated?

Like I say, this is the only behaviour I’ve ever experienced since I started a fresh RetroPie system a few months ago. I.e. it’s not just because I’ve updated to amiberry 5.

**Reply by celeritydesign** [reply]
Great to hear about the GUI toolkit fix by the way!

**Reply by midwan** [reply]
Can you try renaming the `default.uae` file to something else, so it's not picked up automatically? Does that change the behavior regarding the game title after you loaded a WHDLoad game?

**Reply by celeritydesign** [reply]
Yes! That has fixed it, thank you very much!  So anyone else having the same problem, you can do:

cd /opt/retropie/emulators/amiberry/conf/
mv default.uae default.bak

I'm still a bit confused why default.uae was being preselected but not actually being loaded?

I'm guessing the desired behaviour is that amiberry loads default.uae but autofills the config textbox with the gamename?
Or, if default.uae is being used, leave that in the textbox, but add a labelled "Use game name" which when clicked autofills the textbox with the correct filename?

Getting default.uae to load would be helpful to me, as I'd like to enable scanlines for all games by default :-)

**Reply by midwan** [reply]
The `default.uae` is loaded if found, but with WHDLoad titles there's an extra step that sets the configuration based on the XML, if a match is found. So your settings are overridden, essentially.

If you want to configure something by default, it's best to do it in the `amiberry.conf` file, instead. There are several options there that control the defaults, including Line Mode. Take a look at the relevant wiki: https://github.com/midwan/amiberry/wiki/Amiberry.conf-options

**Reply by celeritydesign** [reply]
That’s so annoying. I didn’t read the docs, so I’d set line mode to “scanlines” instead of “2”, to match the game config value :-)
Thanks for the doc link!
